src/applications/vaos/containers/PendingAppointmentPage.jsx

```diff
@@ -0,0 +1,155 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Link } from 'react-router';
+import { connect } from 'react-redux';
+import moment from 'moment';
+import LoadingIndicator from '@department-of-veterans-affairs/formation-react/LoadingIndicator';
+import AlertBox from '@department-of-veterans-affairs/formation-react/AlertBox';
+import { focusElement } from 'platform/utilities/ui';
+import { fetchPendingAppointments } from '../actions/appointments';
+import { FETCH_STATUS, TIME_TEXT, PURPOSE_TEXT } from '../utils/constants';
+
+function formatDate(date) {
+  const parsedDate = moment(date, 'MM/DD/YYYY');
+
+  if (!parsedDate.isValid()) {
+    return '';
+  }
+
+  return parsedDate.format('MMMM D, YYYY');
+}
+
+function formatTimeToCall(timeToCall) {
+  if (timeToCall.length === 1) {
+    return timeToCall[0].toLowerCase();
+  } else if (timeToCall.length === 2) {
+    return `${timeToCall[0].toLowerCase()} or ${timeToCall[1].toLowerCase()}`;
+  }
+
+  return `${timeToCall[0].toLowerCase()}, ${timeToCall[1].toLowerCase()}, or ${timeToCall[2].toLowerCase()}`;
+}
+
+export class PendingAppointmentPage extends React.Component {
+  componentDidMount() {
+    this.props.fetchPendingAppointments();
```

Comment:
Should we check the state first to see if the appointment has already been fetched before calling `fetchPendingAppointments()`?

Reply:
That logic is in the action creator. I went back and forth on where to put it, but I think this way is a little easier to manage because we can reset the loading status in a reducer based on actions that happen and not have to mess with putting that logic in different components.